lib/spack/spack/main.py

```diff
@@ -547,6 +547,13 @@ def setup_main_options(args):
     # Assign a custom function to show warnings
     warnings.showwarning = send_warning_to_tty
 
+    if sys.version_info[:2] == (2, 7):
+        msg = (
+            "Running Spack using Python v2.7 is deprecated and support for it will be "
+            " removed in Spack v0.20, use Python v3.6+ instead."
+        )
+        warnings.warn(msg)
```

Comment:
This is really long -- I think it's better as:

```suggestion
        warnings.warn("Python 2.7 support is deprecated and will be removed in Spack v0.20.")
        warnings.warn("Please move to Python 3.6 or higher.")
```

or:

```suggestion
        warnings.warn(
            "Python 2.7 support is deprecated and will be removed in Spack v0.20.\n"
            "    Please move to Python 3.6 or higher."
        )
```